data_collection/ml_model.py

```diff
@@ -0,0 +1,57 @@
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LinearRegression
+from sklearn.metrics import r2_score, mean_squared_error
+
+
+def train_model(data: dict):
+    """
+    Train a linear regression model using the collected LST, land use, and vegetation indices data.
+
+    Args:
+        data (dict): Dictionary containing 'LST', 'land_use', 'vegetation' keys.
+
+    Returns:
+        model (LinearRegression): Trained linear regression model.
+        metrics (dict): Dictionary with 'MSE' and 'R2' metrics.
+    """
+
+    # Extract the input features and target variable
+    try:
+        lst_data = data.get("LST")
+        land_use_data = data.get("Land_Use")
+        vegetation_data = data.get("Vegetation")
+
+        if not all([lst_data, land_use_data, vegetation_data]):
+            raise ValueError("Missing data. Ensure LST, Land_Use, and Vegetation data are included.")
+
+        # Preparing the features (X) and target (y)
+        # For simplicity, using mode for land use and mean values for NDVI/EVI as a single feature set
+
+        land_use = land_use_data.getInfo().get("LC_Type1")
+        ndvi = vegetation_data[1].getInfo().get("NDVI")
+        evi = vegetation_data[1].getInfo().get("EVI")
```

Comment:
as we saw on the call: `getInfo()` is expensive so call once (cache or use intermediate variable)